apps/condo/domains/ticket/schema/Ticket.js

```diff
@@ -374,12 +381,25 @@ const Ticket = new GQLListSchema('Ticket', {
 
             if (userType === RESIDENT && operation === 'create') {
                 overrideTicketFieldsForResidentUserType(context, resolvedData)
-                const contact = await getOrCreateContact(context, resolvedData)
-                resolvedData.contact = contact.id
                 await setSectionAndFloorFieldsByDataFromPropertyMap(context, resolvedData)
                 setClientNamePhoneEmailFieldsByDataFromUser(get(context, ['req', 'user']), resolvedData)
             }
 
+            if (!resolvedContact) {
+                const resolvedIsResidentTicket = get(resolvedData, 'isResidentTicket')
+                const existedIsResidentTicket = get(existingItem, 'isResidentTicket')
+                const isResidentTicket = isUndefined(resolvedIsResidentTicket) ? existedIsResidentTicket : resolvedIsResidentTicket
+                if (!resolvedContact && isResidentTicket) {
+                    const contact = await getOrCreateContactByClientData(context, resolvedData, existingItem)
+                    resolvedData.contact = contact.id
+                }
+                if (existedIsResidentTicket && resolvedIsResidentTicket === false) {
```

Comment:
please make sure that both conditions won't fire consequently, or you'll get created contact that is not connected to anywhere.